src/sparse_blas/function_table.hpp

```diff
@@ -79,6 +79,15 @@ typedef struct {
     FOR_EACH_FP_AND_INT_TYPE(DEFINE_SET_CSR_DATA);
 
     // optimize_*
+    sycl::event (*optimize_gemm_1_input)(sycl::queue &queue, oneapi::mkl::transpose transpose_A,
+                                         oneapi::mkl::sparse::matrix_handle_t handle,
+                                         const std::vector<sycl::event> &dependencies);
+    sycl::event (*optimize_gemm_2_inputs)(sycl::queue &queue, oneapi::mkl::transpose transpose_A,
```

Comment:
Not trying to nitpick here, but maybe simply naming them as `optimize_gemm_v1` and `optimize_gemm_v2` would be fine. The `_input` flag is a bit misleading when other `optimize_*` functions don't have that in the name. I won't insist on that though, since it's just 4 lines of code in total.

Reply:
That's fine with me, it's done in https://github.com/oneapi-src/oneMKL/pull/403/commits/94b8b8e73f279d2f35887bce1f0c77116496fa43